todoCLI/todoCLI.js

```diff
@@ -0,0 +1,153 @@
+const rl = require(`readline`);
+const fs = require(`fs`);
+
+const menu= `(V) View ✏︎ (n) New ✏︎ (cX) Complete  ✏︎  (dX) Delete ✏︎ (q) Quit\n`;
+
+console.log(`\n\nWelcome to Todo CLI!\n----------------------`);
+const interface = rl.createInterface ( {
+	input: process.stdin,
+	output: process.stdout
+});
+
+waitingForUser();
+
+function waitingForUser () {
+  interface.question(menu, (command) => {
+    let number = parseFloat(command.replace(/[cd]/ , ''))
+    command = command.replace(number, '')
+    switch (command) {
+      case `q`:
+        process.exit();
+        break;
+      case `V`:
+        viewToDoList();
+        break;
+      case `n`:
+        addTask();
+        break;
+      case `c`:
+        completeTask(number);
+        break; 
+      case `d`:
+        deleteTask(number);
+        // console.log(`You typed dX`);
+        break;
+      default:
+        console.log(`Invalid entry\n`);
+        waitingForUser();
+        break;
+      }
+  });
+}
+
+function viewToDoList () {
+  fs.readFile('./toDoList.json', (err, json) => {
+    if (err) { 
+      console.log(`\n   ¯\\\_(ツ)_/¯      You haven't started a list yet. Type 'n' to start one\n`)
+      // waitingForUser();
+    } 
+    else {
+      tasks = JSON.parse(json).tasks;
+      for (let i = 0; i < tasks.length; i++) {
+        const task = tasks[i];
+        if (task.status) { 
+          console.log(`${i+1} [ X ] ${task.description}`);
+        } else {
+          console.log(`${i+1} [   ] ${task.description}`);
+        }
+      }
+      // waitingForUser();
+    }
+    waitingForUser();
+  })
+}
+
+function addTask () {
+  interface.question(`Enter new task: `, (task) => {
+    if (task === "") { 
+      console.log(`Can't add an empty task, please enter task`)
+      waitingForUser()
+    } else {
+      let newTask = {"description": task, status: false}
+      fs.readFile(`./toDoList.json`, (err, json) => {
+        let myToDoList = { "tasks" :[]};
+        if (err) {
+          myToDoList.tasks.push(newTask)
+          writeJSON(myToDoList)
+        } else {
+          myToDoList.tasks = JSON.parse(json).tasks;
+          myToDoList.tasks.push(newTask)
+          writeJSON(myToDoList);
+        }
+        console.log(`                           . . . . . . task added uccessfuly!!!\n`)
+      })
+    }
+    // waitingForUser()
+  })
+}
+
+function writeJSON (myToDoList) {
+  JSONlist = JSON.stringify(myToDoList);
+  fs.writeFile(`toDoList.json`, JSONlist, (err) => {
+    if(err) {
+      console.log(`Could not write file ${err}`);
+    } else {
+      // console.log(`**********\nTo do list updated!!\n`);
+      waitingForUser();
+    }
+  })
+}
+
+function completeTask (num) {
+  interface.question(`Are you sure you want to complete task ${num}? Type 'y' or 'n': `, (confirmation) => {
+    if (confirmation.includes('y')) {
+      fs.readFile('./toDoList.json', (err, json) => {
```

Comment:
Right now you're re-reading the file every time on every command, which is probably not necessary (imagine what happens if you have a _really_ long to-do list). You can probably just read the file once, when you load the program, and then either write it out at the end of the `waitingForUser` (a.k.a. `main()`) loop, or once at the very end when you `q` out of the process (this would, however, have the risk of losing your updates to the list should the program crash before finishing).

Reply:
Unless explicitly told to use asynchronous functions, you might want to use `fs.writeFileSync` and `fs.readFileSync` for now til you get a better hang of JS and asynchronous functions (which can be a PITA)